pkg/predicate/predicate.go

```diff
@@ -202,3 +205,197 @@ func AddTypePredicate(extensionType string, predicates []predicate.Predicate) []
 	preds = append(preds, HasType(extensionType))
 	return append(preds, predicates...)
 }
+
+// MachineStatusHasChanged is a predicate deciding wether the status of a MCM's Machine has been changed.
+func MachineStatusHasChanged() predicate.Predicate {
+	statusHasChanged := func(oldObj runtime.Object, newObj runtime.Object) bool {
+		oldMachine, ok := oldObj.(*machinesv1alpha1.Machine)
+		if !ok {
+			return false
+		}
+		newMachine, ok := newObj.(*machinesv1alpha1.Machine)
+		if !ok {
+			return false
+		}
+		oldStatus := oldMachine.Status
+		newStatus := newMachine.Status
+
+		//Check the Node
+		if oldStatus.Node != newStatus.Node {
+			return true
+		}
+
+		//Check the CurrentStatus
+		oldCurrentStatus := oldStatus.CurrentStatus
+		newCurrentStatus := newStatus.CurrentStatus
+
+		if oldCurrentStatus.Phase != newCurrentStatus.Phase ||
+			oldCurrentStatus.TimeoutActive != newCurrentStatus.TimeoutActive ||
+			newCurrentStatus.LastUpdateTime.After(oldCurrentStatus.LastUpdateTime.Time) {
+			return true
+		}
+
+		//Check the LastOperation
+		oldLastOperation := oldStatus.LastOperation
+		newLastOperation := newStatus.LastOperation
+
+		if oldLastOperation.Type != newLastOperation.Type ||
+			oldLastOperation.State != newLastOperation.State ||
+			oldLastOperation.Description != oldLastOperation.Description ||
+			newLastOperation.LastUpdateTime.After(newLastOperation.LastUpdateTime.Time) {
```

Comment:
```suggestion
			newLastOperation.LastUpdateTime.After(oldLastOperation.LastUpdateTime.Time) {
```

You can use equality.Semantic.DeepEqual(oldLastOperation, newLastOperation)